tests/unit/test_lifecycle.py

```diff
@@ -28,6 +28,23 @@
 pytestmark = pytest.mark.usefixtures("reset_callbacks")
 
 
+@pytest.fixture
```

Comment:
Some of the changes in this file seem unrelated to the environment changes?

Reply:
it's a consequence. This fix implies actually having a `Project` method for getting the YAML loaded as a dict. As a result, I've changed the `return` type of our current project handling functions and thus that cascaded into a change to many unrelated tests

Reply:
for example: https://github.com/canonical/craft-parts/blob/ed49f727e60447d491f7d7b58b03711c2e6a7bfd/craft_parts/executor/environment.py#L204 expects a Dict, but we use pydantic for project validation, which mean we end up packing the ROCK with all the project info under a class object.

Reply:
No probs! :+1: